src/plugins/intel_cpu/tests/unit/ngraph_transformations/arm/convert_reduce_multi_axis.cpp

```diff
@@ -0,0 +1,182 @@
+        function = createInitGraph<ngraph::opset1::ReduceProd>(param);
+        ov::pass::Manager manager;
+        manager.register_pass<ConvertReduceProd>();
+        manager.run_passes(function);
+    }
+    {
+        function_ref = createRefGraph<ngraph::opset1::ReduceProd>(static_param_shape);
+    }
+    auto res = compare_functions(function, function_ref);
+    ASSERT_TRUE(res.first) << res.second;
+}
+
+TEST(TransformationTests, CheckConvertReduceMinTransformationIsNotAppliedForDynaimcShapes) {
+    std::shared_ptr<ov::Model> function(nullptr), function_ref(nullptr);
+    {
+        auto param = std::make_shared<ngraph::opset1::Parameter>(ngraph::element::f32, dynamic_param_shape);
+        function = createInitGraph<ngraph::opset1::ReduceMin>(param);
+        ov::pass::Manager manager;
+        manager.register_pass<ConvertReduceMin>();
+        manager.run_passes(function);
+    }
+    {
+        auto param = std::make_shared<ngraph::opset1::Parameter>(ngraph::element::f32, dynamic_param_shape);
+        function_ref = createInitGraph<ngraph::opset1::ReduceMin>(param);
+    }
+    auto res = compare_functions(function, function_ref);
+    ASSERT_TRUE(res.first) << res.second;
+}
+
+TEST(TransformationTests, CheckConvertReduceMaxTransformationIsNotAppliedForDynaimcShapes) {
+    std::shared_ptr<ov::Model> function(nullptr), function_ref(nullptr);
+    {
+        auto param = std::make_shared<ngraph::opset1::Parameter>(ngraph::element::f32, dynamic_param_shape);
+        function = createInitGraph<ngraph::opset1::ReduceMax>(param);
+        ov::pass::Manager manager;
+        manager.register_pass<ConvertReduceMax>();
+        manager.run_passes(function);
+    }
+    {
+        auto param = std::make_shared<ngraph::opset1::Parameter>(ngraph::element::f32, dynamic_param_shape);
+        function_ref = createInitGraph<ngraph::opset1::ReduceMax>(param);
+    }
+    auto res = compare_functions(function, function_ref);
+    ASSERT_TRUE(res.first) << res.second;
+}
+
+TEST(TransformationTests, CheckConvertReduceSumTransformationIsNotAppliedForDynaimcShapes) {
+    std::shared_ptr<ov::Model> function(nullptr), function_ref(nullptr);
+    {
+        auto param = std::make_shared<ngraph::opset1::Parameter>(ngraph::element::f32, dynamic_param_shape);
+        function = createInitGraph<ngraph::opset1::ReduceSum>(param);
+        ov::pass::Manager manager;
+        manager.register_pass<ConvertReduceSum>();
+        manager.run_passes(function);
+    }
+    {
+        auto param = std::make_shared<ngraph::opset1::Parameter>(ngraph::element::f32, dynamic_param_shape);
+        function_ref = createInitGraph<ngraph::opset1::ReduceSum>(param);
+    }
+    auto res = compare_functions(function, function_ref);
+    ASSERT_TRUE(res.first) << res.second;
+}
+
+TEST(TransformationTests, CheckConvertReduceProdTransformationIsNotAppliedForDynaimcShapes) {
```

Comment:
It seems all this duplication can be avoided using
https://github.com/google/googletest/blob/main/docs/advanced.md#typed-tests
Which makes sense, since the transformation itself and the functions you are using to create the models in scope of this test are also type templates.

Reply:
Applied typed tests approach